lib/mockstrategies/OrderedMockStrategy.js

```diff
@@ -0,0 +1,62 @@
+const expect = require('unexpected')
+  .clone()
+  .use(require('unexpected-messy'));
+const messy = require('messy');
+
+const errors = require('../errors');
+const resolveExpectedRequestProperties = require('../resolveExpectedRequestProperties');
+const trimMessyHeaders = require('../trimMessyHeaders');
+
+module.exports = class OrderedMockStrategy {
+  constructor(requestDescriptions) {
+    this.requestDescriptions = requestDescriptions;
+    this.nextRequestDescriptionIndex = 0;
+  }
+
+  get isEmpty() {
+    return this.nextRequestDescriptionIndex >= this.requestDescriptions.length;
+  }
+
+  firstDescriptionRemaining() {
+    return this.nextDescriptionForIncomingRequest();
+  }
+
+  nextDescriptionForIncomingRequest(requestStruct) {
```

Comment:
The word "next" reads a little funny here now that the strategy could do anything, but I don't really have a better idea :)

Also, it seems like the function could be simplified with `async`/`await`, avoiding the `Promise.resolve()` stuff.